src/fs/metadata_log.cc

```diff
@@ -101,18 +100,51 @@ void metadata_log::write_update(inode_info::file& file, inode_data_vec new_data_
     } else {
         cut_out_data_range(file, new_data_vec.data_range);
     }
-
     file.data.emplace(new_data_vec.data_range.beg, std::move(new_data_vec));
 }
 
 void metadata_log::cut_out_data_range(inode_info::file& file, file_range range) {
-    file.cut_out_data_range(range, [](inode_data_vec data_vec) {
-        (void)data_vec; // TODO: for compaction: update used inode_data_vec
+    using CR = inode_info::file::cut_result;
+    file.cut_out_data_range(range, [&](inode_data_vec data_vec, CR cr) {
+        int new_writes_cnt = 0;
+        switch (cr) {
+        case CR::BOTH:
+            new_writes_cnt++;
+        case CR::LEFT:
+        case CR::RIGHT:
+            new_writes_cnt++;
+        case CR::NONE:
+            break;
+        }
+        std::visit(overloaded {
+            [&](inode_data_vec::in_mem_data& mem) {
+                size_t log_decrease = ondisk_entry_size(ondisk_small_write_header{0,0,static_cast<uint16_t>(data_vec.data_range.size()),0});
+                _rewrite_log_size -= log_decrease;
+                _rewrite_log_size += new_writes_cnt * ondisk_entry_size(ondisk_small_write_header{});
+            },
+            [&](inode_data_vec::on_disk_data& disk_data) {
+                // TODO: Diffrenciate between large write and large write without mtime
+                size_t log_decrease = data_vec.data_range.size() == _cluster_size ?
+                    ondisk_entry_size(ondisk_large_write{}) :
+                    ondisk_entry_size(ondisk_medium_write{});
+                _rewrite_log_size -= log_decrease;
+                // Remaining data is smaller than original so it's less than cluster size
+                _rewrite_log_size += new_writes_cnt * ondisk_entry_size(ondisk_medium_write{});
+            },
+            [&](inode_data_vec::hole_data&) {
+            },
+        }, data_vec.data_location);
     });
 }
 
 inode_info& metadata_log::memory_only_create_inode(inode_t inode, bool is_directory, unix_metadata metadata) {
     assert(_inodes.count(inode) == 0);
+    size_t log_increase = ondisk_entry_size(ondisk_create_inode{});
+    _ondisk_log_size += log_increase;
+    _rewrite_log_size += log_increase;
+    // During rewrite we'll start with a truncate setting the file size to remember trailing holes
+    if (not is_directory)
```

Comment:
following Avi's requests, let's use good old `!` instead of `not` in new code, so there'll be less to fix later